### PR TITLE
Toast extensions on Context

### PR DESCRIPTION
**Pull request by simonschiller**
See #143

### CODE REVIEW

**Comment by simonschiller** [comment]
I signed it!

File: build.gradle [hunk 1]
```diff
@@ -63,6 +63,7 @@ dependencies {
 
     androidTestImplementation 'junit:junit:4.12'
     androidTestImplementation "com.android.support:support-annotations:${versions.supportLibrary}"
+    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.1'
```
Comment (shaishavgandhi05) [inline build.gradle]: Extract version to project build.gradle?

Comment (simonschiller) [reply build.gradle]: Currently, only versions that are used multiple times are extracted (`kotlin`, `supportLibrary` and `atsl`), I wanted to keep consistent with that.

Comment (romainguy) [reply build.gradle]: Before we merge this, @JakeWharton do you have any problem with adding espresso?

Comment (JakeWharton) [reply build.gradle]: Yep. It's a bit overkill though as we could inspect the properties of the returned `Toast` object.

File: src/main/java/androidx/widget/Toast.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import android.content.Context
+import android.support.annotation.StringRes
+import android.widget.Toast
+
+/**
+ * Creates and shows a [Toast] with the given [text]
+ *
+ * @param duration Toast duration, defaults to [Toast.LENGTH_SHORT]
+ */
+fun Context.toast(text: CharSequence, duration: Int = Toast.LENGTH_SHORT): Toast {
```
Comment (JakeWharton) [inline src/main/java/androidx/widget/Toast.kt]: inline

File: src/main/java/androidx/widget/Toast.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    return Toast.makeText(this.applicationContext, text, duration).apply { show() }
+}
+
+/**
+ * Creates and shows a [Toast] with text from a resource
+ *
+ * @param resId Resource id of the string resource to use
+ * @param duration Toast duration, defaults to [Toast.LENGTH_SHORT]
+ */
+fun Context.toast(@StringRes resId: Int, duration: Int = Toast.LENGTH_SHORT): Toast {
```
Comment (JakeWharton) [inline src/main/java/androidx/widget/Toast.kt]: inline

File: src/main/java/androidx/widget/Toast.kt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    return Toast.makeText(this.applicationContext, resId, duration).apply { show() }
```
Comment (JakeWharton) [inline src/main/java/androidx/widget/Toast.kt]: Why the application context? Just use `this`.

File: src/main/java/androidx/widget/Toast.kt [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import android.support.annotation.StringRes
+import android.widget.Toast
+
+/**
+ * Creates and shows a [Toast] with the given [text]
+ *
+ * @param duration Toast duration, defaults to [Toast.LENGTH_SHORT]
+ */
+fun Context.toast(text: CharSequence, duration: Int = Toast.LENGTH_SHORT): Toast {
+    return Toast.makeText(this.applicationContext, text, duration).apply { show() }
```
Comment (JakeWharton) [inline src/main/java/androidx/widget/Toast.kt]: Why the application context? Just use `this`.

Comment (romainguy) [reply src/main/java/androidx/widget/Toast.kt]: Doh, good catch. Yeah don't use the application context.

File: src/main/java/androidx/widget/Toast.kt (hunk 4 above)
Comment (romainguy) [reply src/main/java/androidx/widget/Toast.kt]: Toast doesn't have a `getText()` only a `getView()`, which could make the inspection brittle.

Comment (JakeWharton) [reply src/main/java/androidx/widget/Toast.kt]: Espresso is doing a DFS for TextView's whose text matches the supplied string which would be pretty easy to replicate.

Comment (romainguy) [reply src/main/java/androidx/widget/Toast.kt]: 👍

**Comment by simonschiller** [comment]
Updated the PR @JakeWharton @romainguy

**Comment by JakeWharton** [comment]
Looks great. Thanks!

**Pull request merged**